lib/sync/datasetClientsCleaner.js

```diff
@@ -0,0 +1,78 @@
+var util = require('util');
+var parseDuration = require('parse-duration');
+var async = require('async');
+var _ = require('underscore');
+var DatasetClient = require('./DatasetClient');
+var syncUtil = require('./util');
+
+var storage;
+
+/**
+ * Constructor for DatasetClientsCleaner.
+ * 
+ * A DatasetClientsCleaner will check the status of all the dataset clients in the db, and if any of them hasn't been accessed for the past <retentionPeriod> time, it will remove them from the db to save space.
+ * By default, it will remove dataset clients that haven't been active for 1 day and it will run the check every hour.
+ * Developer can change those parameters via the configuration.
+ * 
+ * @param {Object} opts configuration options for the DatasetClientsCleaner. 
+ * @param {String} retentionPeriod the maximum time the dataset client can be kept in db after it's last access tim. Support format: https://www.npmjs.com/package/parse-duration. Default is '24h'.
+ * @param {String} checkFrequency how often the checks should be running. Default is '1h'.
+ */
+function DatasetClientsCleaner(opts) {
+  this.retentionPeriod = opts.retentionPeriod || '24h';
+  this.checkFrequency = opts.checkFrequency || '1h';
+  this.job;
+}
+
+function cleanDatasetClients(retentionPeriod) {
+  async.waterfall([
+    async.apply(storage.listDatasetClients),
+    function filterDatasetClients(datasetClients, callback) {
+      var datasetClientsToRemove = _.filter(datasetClients, function(datasetClientJson){
+        var datasetClient = DatasetClient.fromJSON(datasetClientJson);
+        return datasetClient.shouldBeRemoved(parseDuration(retentionPeriod));
```

Comment:
To avoid `parseDuration(retentionPeriod)` being called multiple times (once per dataset client) it could probably be moved outside the `_.filter` or even the `waterfall` at the top. I'm assuming the result of that function is deterministic?